ts/Stock/Indicators/DisparityIndex/DisparityIndexIndicator.ts

```diff
@@ -0,0 +1,282 @@
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type {
+    DisparityIndexOptions,
+    DisparityIndexParamsOptions
+} from './DisparityIndexOptions';
+import type DisparityIndexPoint from './DisparityIndexPoint';
+import type IndicatorValuesObject from '../IndicatorValuesObject';
+import type LineSeries from '../../../Series/Line/LineSeries';
+import RequiredIndicatorMixin from '../../../Mixins/IndicatorRequired.js';
+import SeriesRegistry from '../../../Core/Series/SeriesRegistry.js';
+const {
+    seriesTypes: {
+        ema: EMAIndicator,
+        dema: DEMAIndicator,
+        sma: SMAIndicator,
+        tema: TEMAIndicator,
+        wma: WMAIndicator
+    }
+} = SeriesRegistry;
+import U from '../../../Core/Utilities.js';
+const {
+    correctFloat,
+    extend,
+    isArray,
+    merge
+} = U;
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/**
+ * The Disparity Index series type.
+ *
+ * @private
+ * @class
+ * @name Highcharts.seriesTypes.disparityindex
+ *
+ * @augments Highcharts.Series
+ */
+class DisparityIndexIndicator extends SMAIndicator {
+    /**
+     * Disparity Index.
+     * This series requires the `linkedTo` option to be set and should
+     * be loaded after the `stock/indicators/indicators.js` file.
+     *
+     * @sample stock/indicators/disparity-index
+     *         Disparity Index indicator
+     *
+     * @extends      plotOptions.sma
+     * @since        next
+     * @product      highstock
+     * @excluding    allAreas, colorAxis, joinBy, keys, navigatorOptions,
+     *               pointInterval, pointIntervalUnit, pointPlacement,
+     *               pointRange, pointStart, showInNavigator, stacking
+     * @requires     stock/indicators/indicators
+     * @requires     stock/indicators/disparity-index
+     * @optionparent plotOptions.disparityindex
+     */
+    public static defaultOptions: DisparityIndexOptions = merge(SMAIndicator.defaultOptions, {
+        params: {
+            /**
+             * The average used to calculate the Disparity Index indicator.
+             * By default it uses SMA. To use other averages, e.g. EMA,
+             * the stock/indicators/ema.js file needs to be loaded.
+             *
+             * If value is different than ema|dema|tema|wma, then sma is used.
+             */
+            average: 'sma',
+            index: 3,
+            period: 14
+        },
+        marker: {
+            enabled: false
+        },
+        dataGrouping: {
+            approximation: 'averages'
+        }
+    } as DisparityIndexOptions);
+
+    /* *
+     *
+     *  Properties
+     *
+     * */
+
+    public data: Array<DisparityIndexPoint> = void 0 as any;
+    public options: DisparityIndexOptions = void 0 as any;
+    public points: Array<DisparityIndexPoint> = void 0 as any;
+
+    /* *
+     *
+     *  Functions
+     *
+     * */
+
+    public init(this: DisparityIndexIndicator): void {
+        const args = arguments,
+            ctx = this, // Disparity Index indicator
```

Comment:
Why `ctx`?

Reply:
`ctx` means "context". In this case, the main context is the Disparity Index indicator. The variable is not called `indicator` because this name is used later in
```js
function (indicator: Highcharts.Indicator): undefined {
    indicator.prototype.init.apply(ctx, args);
    return;
}
```

It's not my idea/pattern, but it's named same way in every indicator that uses `IndicatorRequired.js`. If I'm not wrong, you did exactly the same in your indicator :) 

![image](https://user-images.githubusercontent.com/23310742/113846955-49dce000-9797-11eb-9380-26f6b7d19fa2.png)

But I also added a comment that indicates the context `// Disparity Index indicator`

Reply:
Yes, I did it exactly the same, but as you can see I didn't think about that as you!
Thanks for explaining!